earth_enterprise/src/fusion/autoingest/sysman/khAssetManager.cpp

```diff
@@ -990,7 +990,6 @@ khAssetManager::MercatorMapDatabaseModify(const MapDatabaseEditRequest &req) {
 
 void
 khAssetManager::GetCurrTasks(const std::string &dummy, TaskLists &ret) {
-  assert(!mutex.TryLock());
   notify(NFY_INFO, "GetCurrTasks");
   khLockGuard lock(theResourceManager.mutex);
```

Comment:
We should move the timedLock logic here so that we can still report a sensible error back to system manager during the times that this lock is held for a long time.